stride/problem/acquisitions.py

```diff
@@ -979,6 +980,78 @@ def default(self):
                           sources=[source], receivers=receivers,
                           geometry=self._geometry, problem=self.problem))
 
+    def from_fullwave(self, acquisitionttr, sourcettr=None, **kwargs):
+        """
+        Populates acquisition container with shot and receiver ids as described 
+        by a Fullwave .ttr acquisition file and an optional source .ttr file 
+        describing the wavelets for each shot. Assumes that each source id is shot
+        individually
+        
+        Parameters
+        ----------
+        acquisitionttr : Path
+            Path to .ttr acquisition Fullwave file (ideally Observed-Trace.ttr)
+        
+        sourcettr: Path, optional
+            Path to .ttr source Fullwave file. Default None
+            (Method not implemented)
+
+        Returns
+        -------
+
+        """
+
+        assert acquisitionttr.lower().split(".")[-1] == "ttr", "Expected .ttr extension in acquisitionttr but found .%s"%acquisitionttr.lower().split(".")[-1]
+        if sourcettr is not None:
+            assert sourcettr.lower().split(".")[-1] == "ttr", "Expected .ttr extension in sourcettr but found .%s"%sourcettr.lower().split(".")[-1]
+            raise NotImplementedError
+
+        # Read 4-byte binary ttr file to retrieve source ids and correspondent receiver ids
+        with open(acquisitionttr, mode='rb') as file:
+
+            # List to store source and receivers ids
+            sources_ids, receiver_ids = [], []
```

Comment:
This works, but I think it would be much easier to use a dictionary with the source ID as keys and the list of receiver IDs as values. Just an idea, though